mvc/views/view-components.rst

```diff
@@ -21,54 +30,98 @@ New to ASP.NET MVC 6, view components are similar to partial views, but they are
 
-- Deriving from `ViewComponent`.
-- Decorating the class with the ``[ViewComponent]`` attribute, or deriving from a class with the ``[ViewComponent]`` attribute.
-- Creating a class where the name ends with the suffix *ViewComponent*.
+- Deriving from `ViewComponent`
+- Decorating a class with the ``[ViewComponent]`` attribute, or deriving from a class with the ``[ViewComponent]`` attribute
+- Creating a class where the name ends with the suffix *ViewComponent*
 
-Like controllers, view components must be public, non-nested, non-abstract classes.
+Like controllers, view components must be public, non-nested, and non-abstract classes.
 
-Examine the ViewComponent class
---------------------------------
+View component methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-- Examine the *src\\TodoList\\ViewComponents\\PriorityListViewComponent.cs* file:
+- Contain ``Invoke`` or ``InvokeAsync`` methods that return an ``IViewComponentResult``
+- Fully support constructor `Dependency Injection <https://docs.asp.net/en/latest/fundamentals/dependency-injection.html>`__
+- Typically initialize a model and pass it to a view by calling the ``ViewComponent`` ``View`` method. 
+- Parameters come from the calling method, not HTTP, there is no model binding. This allows you to overload ``Invoke``/``InvokeAsync``  (which you can't do with controllers).
+- Do not take part in the controller lifecycle, which means you can't use action filters in a view component
+- Are not reachable directly as an HTTP endpoint, they are invoked from your code (usually in a view). A View Component never handles a request.
+- Are overloaded on the signature rather than any details from the current HTTP request
 
-  .. code-block:: c#
-    :linenos:
+View search path
+^^^^^^^^^^^^^^^^^^
+
+- The default view name is *Default.cshtml*
+- The runtime searches for the view in the following paths:
 
-    using System.Linq;
-    using Microsoft.AspNet.Mvc;
-    using TodoList.Models;
+  - Views/<controller_name>/Components/<view_component_name>
+  - Views/Shared/Components/<view_component_name>
 
-    namespace TodoList.ViewComponents
-    {
-      public class PriorityListViewComponent : ViewComponent
-      {
-        private readonly ApplicationDbContext db;
+Invoking a view component
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-        public PriorityListViewComponent(ApplicationDbContext context)
-        {
-          db = context;
-        }
+To use the view component, call the synchronous ``@Component.Invoke("Name of View Component", <parameters>)`` or asynchronous ``@Component.InvokeAsync("Name of View Component", <parameters>)`` from a view. The parameters will be passed to the ``Invoke`` or ``InvokeAsync`` method with a matching signature.  The ``PriorityList`` view component developed in the article is invoked from the *Views/TodoItems/Index.cshtml* view file. In this example the synchronous ``Invoke`` method is called and 3 is passed to the view component:
```

Comment:
Let's just mention the async version (`Component.InvokeAsync`) since the sync version is going away.